api.js

```diff
@@ -5,6 +5,8 @@
 
 'use strict'
 
+/* eslint sonarjs/cognitive-complexity: ["error", 21] -- TODO: https://issues.newrelic.com/browse/NEWRELIC-5252 */
```

Comment:
shouldn't this be on the actual function that's in violation? it looks like there are actually 2 functions